### PR TITLE
Use GitPython for all git operations

### PR DESCRIPTION
**Pull request by jaindhairyahere**
Moving from using subprocesses to execute git commands to using
git module for the same. This eliminates # nosec dismissals for
some of the commands that shell out.

The tests are updated to accommodate the changes.

Resolves: #619

Signed-off-by: Dhairya Jain <jaindhairya2001@gmail.com>

### CODE REVIEW

**Comment by rnjudge** [comment]
@jaindhairyahere can you remove the first commit from this PR? ("Merge branch 'main' of https://github.com/jaindhairyahere/tern into main")

File: tern/analyze/common.py [hunk 1]
```diff
@@ -252,20 +249,11 @@ def get_git_url(dockerfile_path):
...
-            # pop the last line which is an empty line
-            lines.pop()
-            url_list = set()
-            for line in lines:
-                extract_url = extract_git_url_from_line(line)
-                if extract_url:
-                    url_list.add(extract_url)
-    except (subprocess.CalledProcessError, FileNotFoundError):
+        repo = git.Repo(dockerfile_folder_path, search_parent_directories=True)
+        url_list = {i.url for i in repo.remotes}
```
Comment (rnjudge) [inline tern/analyze/common.py:255]: Won't this return `url_list` as a set instead of a list?

**Comment by rnjudge** [comment]
A few small changes requested. Otherwise, looks good.

**Comment by rnjudge** [comment]
@jaindhairyahere are you still available to update this PR?

**Comment by jaindhairyahere** [comment]
I've made the changes. But the checks aren't passing here, although they passed in my local system

**Comment by rnjudge** [comment]
I'll merge this once I get some clarity upstream on the failing CI tests.

**Pull request merged**